fuzz_lightyear/main.py

```diff
@@ -121,12 +124,14 @@ def run_tests(
     :param seed: used for random generation of test input
     :param ignore_exceptions: if True, ignores HTTP exceptions to requests.
     :param disable_unicode: if True, only use ASCII characters to fuzz strings
+    :param max_fuzz_depth: used for preventing recursion Objects with self reference
     """
     if seed is not None:
         get_settings().seed = seed
     if disable_unicode:
         get_settings().unicode_enabled = False
 
+    get_settings().max_fuzz_depth = max_fuzz_depth
```

Comment:
Do we need to check if this is already set here to not overwrite it?

Reply:
At this point `max_fuzz_depth` will either have the value passed by user as argument or the default value. For the latter case, it'll have no effect as it'll set it to the default value.